hw3_2/hw3-2/Program.cs

```diff
@@ -0,0 +1,452 @@
+                return;
+            }
+
+            path = args[0];
+            argument = args[1];
+            if (argument != "-c" && argument != "-u")
+            {
+                Console.WriteLine(
+                    "Incorrect argument. Enter file path and \"-c\" if you want to compress text\nor \"-u\" if decompress");
+                return;
+            }
+
+            if (!File.Exists(path))
+            {
+                Console.WriteLine("Incorrect file path!");
+                return;
+            }
+
+            Lzw lzw = new();
+            if (argument == "-c")
+            {
+                string? text = "";
+
+                using (StreamReader reader = new StreamReader(path))
+                {
+                    if ((text = await reader.ReadToEndAsync()) == null)
+                    {
+                        Console.WriteLine("Empty file!");
+                        return;
+                    }
+                }
+
+                bool isEmpty;
+                List<Byte[]> listOfCodes = lzw.LzwCompression(text, out isEmpty);
+                if (isEmpty)
+                {
+                    Console.WriteLine("Empty file!");
+                    return;
+                }
+
+                string newName = path + ".zipped";
+                using (FileStream zippedFile = File.Create(newName))
+                {
+                    if (!File.Exists(newName))
+                    {
+                        Console.WriteLine("Incorrect file path!");
+                        return;
+                    }
+
+                    foreach (byte[] code in listOfCodes)
+                    {
+                        zippedFile.Write(code, 0, code.Length);
+                    }
+                }
+
+                Console.WriteLine($"Compression ratio: {4 * listOfCodes.Count / (float)(2 * text.Length)}");
+                return;
+            }
+
+            if (argument == "-u")
+            {
+                Byte[] allBytes;
+                allBytes = File.ReadAllBytes(path);
+                if (allBytes.Length == 0)
+                {
+                    Console.WriteLine("Empty file!");
+                    return;
+                }
+
+                bool isEmpty;
+                bool isCorrect;
+                string decompressedText = lzw.LzwDecompression(GroupBytesByFour(allBytes), out isEmpty, out isCorrect);
+                if (isEmpty)
+                {
+                    Console.WriteLine("Empty file!");
+                    return;
+                }
+
+                if (!isCorrect)
+                {
+                    Console.WriteLine("Impossible to decompress. Wrong codes");
+                }
+
+                string newFileName = path.Replace(".zipped", "");
+
+                StreamWriter unzippedFileWriter = new StreamWriter(newFileName);
+                unzippedFileWriter.Write(decompressedText);
+                unzippedFileWriter.Close();

```

Comment:
```suggestion
                using var unzippedFileWriter = new StreamWriter(newFileName);
                unzippedFileWriter.Write(decompressedText);
```
1) Не пишите тип, когда он очевиден из право части.
2) using тут очень к месту.